### PR TITLE
feat!: new website frontend and app ui

### PR DESCRIPTION
**Pull request by orvn**
Ships new NextJS website including:

- Marketing site
- App UI
- Docs

See commit list for details, since this is an initiative with a large collection of epics

> Note: working on getting the build running correctly with Cloudflare pages

1-home

2-appui

3-faq

### CODE REVIEW

**Comment by orvn** [comment]
Marking as _ready for review_ to trigger CI

File: .github/workflows/website.yml
Comment (olizilla) [inline .github/workflows/website.yml:26]:
```suggestion
```
the step after this one does `npm ci` but adds in caching the dep tree on github based on the state of the package-lock.js
see: https://github.com/bahmutov/npm-install#basic

File: .github/workflows/website.yml
Comment (olizilla) [inline .github/workflows/website.yml:42]:
```suggestion
```

File: packages/website/package.json
Comment (olizilla) [inline packages/website/package.json:8]: Please make `npm start` run the website in dev mode. We don't use an npm run script for production, so we optimise for developers here and make `npm start` do the thing.

File: .github/workflows/website.yml
Comment (orvn) [reply .github/workflows/website.yml]: Got it, makes sense. Removed both of these in favour of `npm i` being run from the repo root, for all workspaces.

File: .github/workflows/website.yml
Comment (olizilla) [inline .github/workflows/website.yml:40]: the `env` property on the next line has been detached from the the run step that needs it above this line

File: packages/website/README.md
Comment (olizilla) [inline packages/website/README.md:58]:
```suggestion
```

File: packages/website/README.md
Comment (olizilla) [inline packages/website/README.md:73]:
```suggestion
```

File: packages/website/README.md
Comment (olizilla) [inline packages/website/README.md:86]:
```suggestion
```

File: packages/website/README.md
Comment (olizilla) [inline packages/website/README.md:97]: The olden text here should still be correct. pls revert.

**Review by olizilla — CHANGES_REQUESTED** [review]
I'm excited to get into the meat of this PR. In the mean time here is a list of boring nitpicks. sorry. The feedback will get better. there is just so many files.

- packages/client/package-lock.json shows as changed, but there is not change to package.json, plz revert.
- packages/website/.github/pull_request_template.md wont apply in monorepo world. plz delete.
- what is packages/website/.legacy ? what will it take to be free of our past?

**Comment by orvn** [comment]
Thanks @olizilla! There are indeed some vintage vestiges—doing a cleanup

**Comment by olizilla** [comment]
The new UI is really sharp! Nice work!

Some notes on things that I hit while trying it out... not blockers but should go on the list of things to fix.

### Odd scroll behaviour on the files list

flipping between tokens and then back to files the page was scrolling to the middle of the list for me on Firefox

nu-web3-scroll-misbhave

i see similar jumpy scrolling when using the files search feature (which is otherwise a very good addition!)

also notable that the list of storage providers needs a max length and a _(+ 5 more)_ style visual limit rather than listing all of them, as you can see in the animation, that list can get looooong:

Screenshot 2022-03-09 at 14 01 16


### Make new token flow clearer

I didn't immediately know how to make a new token, and the first one i made (called `shiney`) ended up in the middle of the tokens list so it wasn't clear which one was new. I would split the create token flow out from the "existing tokens list" view.

nu-web3-new-token-flow

**Comment by orvn** [comment]
Awesome, thanks @olizilla:

> Odd scroll behaviour on the files list

Working on this—it happens because it gets the (incorrect) height of the files table before the file list is fully loaded. Firefox is my default as well, but will do a cross-browser test when this change is up. This might have to be put in its own issue post launch, because I want to make sure it works as expected and with a solution a little better than adding a simple timeout.

> The list of storage providers needs a max length and a (+ 5 more)

Oh, that's good to know, I hadn't seen a case with that many SPs before (I'm guessing this is your local API/DB), fill fix!

> Make new token flow clearer

A couple of people have mentioned that it could be clearer, and we have a more obvious flow that's (almost) done—will PR that into this branch today.

File: packages/docs/.gitmodules
Comment (yusefnapora) [reply packages/docs/.gitmodules:1]: As the originator of this submodule hack, I can say no... it was not worth it. 

@orvn feel free to just copy that repo over if you want to clean things up. If you do, there's a thing in the package.json scripts to make sure you've cloned the submodules before you build, so you can remove that as well.

File: packages/docs/docs/.vuepress/config.js
Comment (yusefnapora) [reply packages/docs/docs/.vuepress/config.js:1]: yeah, you should be able to get rid of the .vuepress dir

File: .github/workflows/website.yml
Comment (olizilla) [inline .github/workflows/website.yml:25]: nit: It's customary round these packages to use the CI Test step to run `npm test` and we configure the `test` run script in each project's package.json to run all the targets that should be run when testing the software, which can include linting.

using `npm-run-all` in the package.json run script definitions is our friend here, as it lets us have specific targets like `lint:ts` and then have the `test` target run all the other targets that make sense like `npm-run-all lint:ts test:widgets coverage etc`

File: packages/client/tsconfig.json
Comment (olizilla) [inline packages/client/tsconfig.json:41]: Suprising! why does the js client's tsconfig need to change for the new website? Can we revert?

File: packages/docs/.gitattributes
Comment (olizilla) [inline packages/docs/.gitattributes:1]: discuss: might make more sense to pull this up to the root of the monorepo if we need it.

File: packages/docs/.github/workflows/algolia.yml
Comment (olizilla) [reply packages/docs/.github/workflows/algolia.yml:5]: @yusefnapora this is to encourage algolia to scrape the site for search super powers, right? any reason not to just update the search index when we change the content on deploy?

File: packages/docs/code-snippets/external/example-image-gallery/src/css/style.css
Comment (olizilla) [inline packages/docs/code-snippets/external/example-image-gallery/src/css/style.css:2]: @yusefnapora try and keep example sites self contained for much CID-based self-contained portability, and *checks notes* _performance_! yep the old theory that _"they user might have loaded this font already from another site so it'll be faster"_ is no more. https://wicki.io/posts/2020-11-goodbye-google-fonts/

File: packages/docs/code-snippets/external/example-image-gallery/src/index.html
Comment (olizilla) [inline packages/docs/code-snippets/external/example-image-gallery/src/index.html:12]: @yusefnapora hot linking alert... this path may not still exist in the new website. 

Consider hot-linking to external resources by CID / gateway url, they, in theory, last longer. Or better still, be self contained so your root CID contains all the things.

File: packages/docs/code-snippets/external/example-image-gallery/src/js/helpers.js
Comment (olizilla) [inline packages/docs/code-snippets/external/example-image-gallery/src/js/helpers.js:66]: @yusefnapora this kind of navigating from JS breaks when sites are used on path-based ipfs gateways like ipfs.io. Your path will be prefixed by the root cid, like /bafy123/settings.html

File: packages/docs/code-snippets/how-to/package.json
Comment (olizilla) [inline packages/docs/code-snippets/how-to/package.json:12]: olden

File: packages/docs/src/theme/Footer/footer.json
Comment (olizilla) [inline packages/docs/src/theme/Footer/footer.json:21]: nit: plz use at least domain relative urls throughout this file here so it's self-contained. I was clicking around a staging build of the docs site and ended up back on the production site, which gets confusing if the two aren't radically different.

I suspect there wont be a way in docusarus to have these be path relative urls, so they will break on path based ipfs gateways, but I can live with that.

File: packages/website/.github/pull_request_template.md
Comment (olizilla) [inline packages/website/.github/pull_request_template.md:2]: A left-over from a before time? This should be removed from this PR, as doesn't make much sense in the monorepo.

File: packages/website/.gitignore
Comment (olizilla) [inline packages/website/.gitignore:2]: @orvn what happened to the robots 🤖 ? Do we still have one?

File: packages/website/assets/icons/github.tsx
Comment (olizilla) [inline packages/website/assets/icons/github.tsx:1]: curious: why this one a `.tsx` ?

File: packages/website/assets/icons/globe.js
Comment (olizilla) [inline packages/website/assets/icons/globe.js:2]: `height={20.004}` sub-sub-sub-pixel rendering! engage!

File: packages/website/assets/icons/infoA.js
Comment (olizilla) [inline packages/website/assets/icons/infoA.js:13]: nit: what happpens here if the users system doesn't have Helvetica? Do svgs fallback in the same way as htmls? what if the users default font is butt-ugly, or has awkward glyph positioning relative to the containgin element? Might be better to convert a question mark to an svg path.

File: packages/website/assets/icons/search.tsx
Comment (olizilla) [inline packages/website/assets/icons/search.tsx:1]: tsx

File: packages/website/assets/icons/windows.js
Comment (olizilla) [inline packages/website/assets/icons/windows.js:5]: nit: suprising id attributes. We've used `svgo` previously to handle that job. This may already be part of the workflow as the others so far look clean.

File: packages/website/assets/illustrations/cross.js
Comment (olizilla) [inline packages/website/assets/illustrations/cross.js:1]: nit: The "Component per image" pattern here leads to *a lot* of trivial components... could a build step create them or could we re-use a single component... or just an  tag in the parent?

File: packages/website/components/accordionblock/accordionblock.js
Comment (olizilla) [inline packages/website/components/accordionblock/accordionblock.js:1]: whao extreme commenting

File: packages/website/components/account/fileUploader/fileUploader.scss
Comment (olizilla) [inline packages/website/components/account/fileUploader/fileUploader.scss:35]: why not a css border?

File: packages/website/pages/api/user/[...tokens].js
Comment (olizilla) [inline packages/website/pages/api/user/[...tokens].js:1]: surprising filename `[...tokens].js` !?

Comment (olizilla) [reply packages/website/pages/api/user/[...tokens].js:1]: oh my it is just how next.js api dynamic routes work https://nextjs.org/docs/api-routes/dynamic-api-routes

Comment (olizilla) [reply packages/website/pages/api/user/[...tokens].js:1]: :grandad:

**Comment by olizilla** [comment]
Most of my comments so far are nitpicks that don't need to block this PR from landing if we are keen to get it live asap.

The only blocker currently is the `.legacy` directory. To cut over to the new site we should be confident that we have all the functionality from the old site ported over and the old site should be removed.

**Comment by orvn** [comment]
> Most of my comments so far are nitpicks that don't need to block this PR from landing if we are keen to get it live asap.
> 
> The only blocker currently is the `.legacy` directory. To cut over to the new site we should be confident that we have all the functionality from the old site ported over and the old site should be removed.

Cool, removed that directory! It was just there for easy reference without jumping between commit hashes. The only thing that I do see missing comparing the current site and this site is the link to Filfox URLs for each Filecoin SP.

File: packages/website/pages/404.js
Comment (olizilla) [inline packages/website/pages/404.js:7]: @orvn do we track 404s in countly somewhere else? Looks like it's been removed from this 404 page.

File: packages/client/tsconfig.json
Comment (olizilla) [reply packages/client/tsconfig.json:41]: @orvn plz revert or explain here

File: packages/website/.gitignore
Comment (olizilla) [reply packages/website/.gitignore:2]: @orvn plz give background on robots.txt removal. Is deliberate or lost in the refactor?

**Comment by olizilla** [comment]
## Notes

It looks lovely and the attention to detail is great. I like the pattern of splitting out the content as .json files, makes it easier to edit copy and see the _words for the ~~trees.~~ tags._

Visually, it's a delight. I have a concern that logged in view of the app doesn't give enough space and focus to the users uploaded file listing, which i see as the main thing for this service. The file list feels tight / constrained and a little uncomfortable to use right now. We can iterate on this.

I have a non-blocking issue, and I feel ridiculous raising this, but I'm a little sad that the linting rules for the website are diverging further from those of the rest of the project. It doesn't feel like the JS in this project is trying to fit in with the prevailing style of the rest. This is a `standard` house. One to discuss over a virtual beverage after this lands.

## Questions

- Where do the Zero Components come from?

## TODO

at a minimum we still need to fix these snags or explain why they are needed

- [ ] revert change to packages/client/package-lock.json
- [ ] revert change to packages/client/tsconfig.json
- [ ] delete packages/website/.github/pull_request_template.md
- [ ] remove packages/website/.legacy

you will win more of my gratitude and affection for responding to or fixing other snags commented inline before merging.

**Review by alanshaw — CHANGES_REQUESTED** [review]
Unfortunately I can't `npm install` on my M1 Mac:

```console
npm ERR! code 1
npm ERR! path /Users/alan/Code/web3-storage/web3.storage/node_modules/optipng-bin
npm ERR! command failed
npm ERR! command sh -c node lib/install.js
npm ERR! node:internal/child_process:413
npm ERR!     throw errnoException(err, 'spawn');
npm ERR!     ^
npm ERR! 
npm ERR! Error: spawn Unknown system error -86
npm ERR!     at ChildProcess.spawn (node:internal/child_process:413:11)
npm ERR!     at spawn (node:child_process:700:9)
npm ERR!     at /Users/alan/Code/web3-storage/web3.storage/node_modules/bin-check/index.js:22:12
npm ERR!     at /Users/alan/Code/web3-storage/web3.storage/node_modules/executable/index.js:27:4
npm ERR!     at FSReqCallback.oncomplete (node:fs:199:5) {
npm ERR!   errno: -86,
npm ERR!   code: 'Unknown system error -86',
npm ERR!   syscall: 'spawn'
npm ERR! }
```

I'll look into this and maybe come back with more feedback once I get the site running locally.

In the mean time, there's a few things @olizilla has pointed out that **need** to be addressed before merging:

* Website: remove website/.github/pull_request_template.md
* Website: remove .legacy directory
* Website: revert the version number change in package.json
* Website: revert change to packages/client/package-lock.json
* Website: revert change to packages/client/tsconfig.json

Please create an issue (or multiple if you prefer) for the following snags and we’ll tackle them **after the main PR lands**:

* Docs: Algolia - make it run after deploy instead of on a cron
* Docs: Github workflows move to root and ensure are working
* Docs: Remove submodules
* Website: remove all references to yarn - this is an npm project
* Website/Docs: lint with standard for consistency with the other packages
* Website: some typesccript components - please convert to JS so we have a consistent language choice
* Website: Remove verbose commenting. It makes sense in the SCSS to group styles that apply to the same subject but everywhere else it seems to be describing very obvious sections of the code and does not add any value and we don’t do this anywhere else. e.g.
    * `// ===================================================================== Imports`
    * `// ///////////////////////////////////////////////////////////////////// General`
    * `// -----------------------------------------------------------------------------`
* Website: fix regression: 404’s are no longer tracked in countly
* Website: remove images with “-legacy” suffix

**Comment by alanshaw** [comment]
Fix for M1 issue: https://github.com/web3-storage/web3.storage/pull/1120

Additional feedback:

Blocking:

* Docs: does not retain logged in state
* Website: the "about" link does not seem to scroll to the correct place on the page
* Website: maintenance message covers main nav:
    * Screenshot 2022-03-17 at 15 50 34

Later:

* Website: fix regression in `npm run dev` - `.env` in root not used (need to add to `packages/website`). See [here](https://github.com/web3-storage/web3.storage/blob/dc2d3590e243d94d569c38f4a07ec73582a6aafb/packages/website/package.json#L7) for solution
* Docs header nav links are not relative, linking directly to `https://web3.storage/...`

File: packages/website/package.json
Comment (orvn) [reply packages/website/package.json:24]: At one point it was causing some build issues, but only for users running locally on Windows machines. Will try to unpin and have someone test again

File: packages/website/package.json
Comment (orvn) [reply packages/website/package.json]: Ah, got it. Reverted and changed this PR title to `feat!:` format. So I assume this will be the squashed commit title and get incremented by release please as a major version?

**Comment by orvn** [comment]
### Update 

- Blocking cleanup and reverting above is done! (relevant [tasks above](https://github.com/web3-storage/web3.storage/pull/1001#issuecomment-1065046971) are checked)
- For the remaining items and any discussion, I'll make a series of issues (I'll make sure to group them by type so it isn't too many)
  - This includes the ones [described here](https://github.com/web3-storage/web3.storage/pull/1001#pullrequestreview-913026310) 
  - As well as anything else that hasn't been addressed in the code review comments above.


Additionally, the following recent enhancements were merged in (these were not critical for launch, just what we've been working on recently):

- [x] Checkmark appears when an action is performed, clearly indicating that something changed on the table
- [x] Fix to modal skew effect, which was being obscured by overflow
- [x] Ellipsis on upload now blink, indicating progress, so the user doesn't think the app is stuck on large chunked uploads
- [x] A toggle appears of the number of storage providers > 5
- [x] Storage providers are linked to Filfox and have link styles
- [x] Improvements to scrolling after pagination, or when loading a page with GET params, in a variety of cases
- [x] Improvements for IE11 (there are still a lot of issues)
- [x] Browser update banner for IE
- [x] Fix for an issue where the marketing message bar would overlap content at the top of the page, when a long/multiline message was supplied
- [x] Reduced an unintended layout-shift/blinking effect when the table was refreshed by adding a min height
- [x] Optimized the token flow feedback so that it's more clear for users
- [x] Restored the `mailto:` link in the usage bar on the account dashboard and removed the extra one in the footer

File: packages/website/package.json
Comment (alanshaw) [reply packages/website/package.json]: yes exactly

**Review by alanshaw — APPROVED** [review]
Ok so, the big issue I have is that the docs do not retain the logged in state. I realised this yesterday and listed it as a blocker.

It's the primary reason we moved from the subdomain so I'm disappointed it hasn't been considered.

I think we're going to have to launch with this as a known issue. It needs to be addressed with high priority ASAP though.

Many thanks for all the work, it looks incredible 🚀

**Comment by orvn** [comment]
> Ok so, the big issue I have is that the docs do not retain the logged in state. I realised this yesterday and listed it as a blocker.
> 
> It's the primary reason we moved from the subdomain so I'm disappointed it hasn't been considered.
> 
> I think we're going to have to launch with this as a known issue. It needs to be addressed with high priority ASAP though.
> 
> Many thanks for all the work, it looks incredible 🚀

Thanks @alanshaw! Regarding the docs, there wasn't any particularly stable way to integrate them into the Next site in their current form (Docusaurus and the HTTP API in ReDoc). 

After a chat with @yusefnapora, we proceeded with the reskin of all the styles, but didn't really spend time on all the visual details within it. The next (no pun intended) step is to use something like [Nextra](https://twitter.com/rauchg/status/1333137455176851457) to bring the docs into the Next static app (will open a new issue for this).

File: packages/website/components/account/fileUploader/fileUploader.scss
Comment (orvn) [reply packages/website/components/account/fileUploader/fileUploader.scss:35]: I think a dashed border was the original intent, but it looked a little asymmetrical as the screen sized down (probably not a huge impact on the design, will try it out on a few different screen sizes).

File: packages/website/components/accordionblock/accordionblock.js
Comment (orvn) [reply packages/website/components/accordionblock/accordionblock.js:1]: Is there a preferred comment format to delineate sections? @olizilla

File: packages/website/pages/api/user/[...tokens].js
Comment (orvn) [reply packages/website/pages/api/user/[...tokens].js:1]: Yeah! It doesn't actually export in the static site or anything, but this was a nice mock API to easily test some weird values with when running locally.

File: packages/website/pages/404.js
Comment (orvn) [reply packages/website/pages/404.js:7]: No, this wasn't intentional—thanks for catching. We had some issues with the 404 at one point and this particular Countly event needs to be added back (we did a sweep of Countly throughout the site to ensure everything was still in there, but this particular one must have been missed).

File: packages/website/assets/icons/windows.js
Comment (orvn) [reply packages/website/assets/icons/windows.js:5]: SVGs _should_ have all been optimized, and _svgo_ is still included. Will search through and find any unoptimized SVGs (some PNGs might be getting converted to SVGs soon as well, so there may be some new ones soon).

**Pull request merged**
